test/els_progress_SUITE.erl

```diff
@@ -0,0 +1,136 @@
+%%==============================================================================
+%% Tests for the 'progress' functionality
+%%==============================================================================
+-module(els_progress_SUITE).
+
+%%==============================================================================
+%% Common Test Callbacks
+%%==============================================================================
+-export([ suite/0
+        , init_per_suite/1
+        , end_per_suite/1
+        , init_per_testcase/2
+        , end_per_testcase/2
+        , groups/0
+        , all/0
+        ]).
+
+%%==============================================================================
+%% Testcases
+%%==============================================================================
+-export([ sample_job/1
+        , failing_job/1
+        ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include_lib("common_test/include/ct.hrl").
+-include_lib("stdlib/include/assert.hrl").
+
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Types
+%%==============================================================================
+-type config() :: [{atom(), any()}].
+
+%%==============================================================================
+%% CT Callbacks
+%%==============================================================================
+-spec suite() -> [tuple()].
+suite() ->
+  [{timetrap, {seconds, 30}}].
+
+-spec all() -> [{group, atom()}].
+all() ->
+  [{group, tcp}, {group, stdio}].
+
+-spec groups() -> [atom()].
+groups() ->
+  els_test_utils:groups(?MODULE).
+
+-spec init_per_suite(config()) -> config().
+init_per_suite(Config) ->
+  els_test_utils:init_per_suite(Config).
+
+-spec end_per_suite(config()) -> ok.
+end_per_suite(Config) ->
+  els_test_utils:end_per_suite(Config).
+
+-spec init_per_testcase(atom(), config()) -> config().
+init_per_testcase(sample_job = TestCase, Config) ->
+  Task = fun(_) -> ok end,
+  setup_mocks(Task),
+  [{task, Task} | els_test_utils:init_per_testcase(TestCase, Config)];
+init_per_testcase(failing_job = TestCase, Config) ->
+  Task = fun(_) -> exit(fail) end,
+  setup_mocks(Task),
+  [{task, Task} | els_test_utils:init_per_testcase(TestCase, Config)].
+
+-spec end_per_testcase(atom(), config()) -> ok.
+end_per_testcase(TestCase, Config) ->
+  els_test_utils:end_per_testcase(TestCase, Config),
+  teardown_mocks().
+
+%%==============================================================================
+%% Testcases
+%%==============================================================================
+
+-spec sample_job(config()) -> ok.
+sample_job(_Config) ->
+  {ok, Pid} = new_background_job(),
+  wait_for_completion(Pid),
+  ?assertEqual(length(entries()), meck:num_calls(sample_job, task, '_')),
```

Comment:
I think this module is not included in this PR, is it? How can the tests not fail? 😮

Reply:
The `sample_job` is mocked using the `non_strict` option. This allows to define mocks for non-existing modules and functions. Something that I discovered recently with `meck` and makes our life so nice.

Reply:
Magic 🌟